libs/ledger-live-common/src/exchange/providers/swap.ts

```diff
@@ -203,14 +231,47 @@ function transformData(providersData) {
   return transformed;
 }
 
-export const getProvidersData = async () => {
-  const providersData = await network({
-    url:
-      "https://crypto-assets-service.api.ledger.com/v1/partners" +
-      "?output=name,signature,public_key,public_key_curve" +
-      "&service_name=swap",
+export const getProvidersData = async (): Promise<Record<string, ProviderData>> => {
+  const { data: providersData } = await network<ProvidersDataResponse>({
+    method: "GET",
+    url: "https://crypto-assets-service.api.ledger.com/v1/partners",
+    params: {
+      output: "name,signature,public_key,public_key_curve",
+      service_name: "swap",
+    },
   });
-  return transformData(providersData.data);
+
+  return transformData(providersData);
+};
+
+/**
+ * Retrieves the currency data for a given ID
+ * coins and tokens are litteraly the same on our case but we need to call the correct endpoint.
+ * @param currencyId The unique identifier for the currency.
+ * @param type Specifies the type of currency, either "tokens" or "coins".
+ * @returns A promise that resolves to the currency data including ID, serialized config, and signature.
+ */
+export const findExchangeCurrencyData = async (
+  currencyId: string,
+  type: "tokens" | "coins",
+): Promise<CurrencyData> => {
+  const { data: currencyData } = await network<CurrencyDataResponse>({
+    method: "GET",
+    url: `https://crypto-assets-service.api.ledger.com/v1/${type}`,
+    params: {
+      output: "id,exchange_app_config_serialized,exchange_app_signature",
+      id: currencyId,
+    },
+  });
+  // TODO: maybe throw errors/warning if many currencies fetched with the same ID.
+  if (!currencyData.length) {
+    throw new Error(`Currency data not found for id ${currencyId}`);
```

Comment:
Old message is "Exchange, missing configuration for ..."
https://github.com/LedgerHQ/ledger-live/blob/f0721f22c6f1bb662806299415c3a0bcb333db7c/libs/ledger-live-common/src/exchange/index.ts#L60C22-L60C57
-> please add proper i18n support for this error
-> use old message for english

Also, yes, let's have another error is there are 2 currencies with same id. Should not happen because on CAL backend, ids are supposed to be unique. But in case it happens, it's better to throw an error than "just use the first one".

Reply:
Added check to throw error for many ids founds. But no i18n this is LLC